Add Transactions Filtering

cldt-fr opened the pull request:
- Migrate URL to qonto.com (@See qonto.eu becomes qonto.com)
- Add TransactionFilterBuilder.php for easily building query string for fetching transactions
- Add new funtion listFilter in ApiTransactions.php for filtering transactions in a list request
- Add DateUtils.php for validating date parameter in filtering (ISO8601)

INFO803TEST committed:
- Migrate URL to qonto.com (@See qonto.eu becomes qonto.com)
- Add TransactionFilterBuilder.php for easily building query string for fetching transactions
- Add new funtion listFilter in ApiTransactions.php for filtering transactions in a list request
- Add DateUtils.php for validating date parameter in filtering (ISO8601)

ericabouaf commented on Qonto/Model/TransactionFilterBuilder.php:
```diff
@@ -0,0 +1,164 @@
+
+namespace neyric\Qonto\Model;
+
+use Cassandra\Date;
```
Is this used somewhere ?

cldt-fr replied:
Yes for validating date parameter in filtering (ISO8601) in TransactionFilterBuilder.php

ericabouaf reviewed (commented):
Nice !
Just wondering if the `Cassandra\Date` use statement is useful ?

Commit:
Merge branch 'ericabouaf:master' into master

Commit:
- Add ExternalTransfer.php (new features from the Qonto API)
- List, List filters and Create an ExternalTransfer.php
- Add "X-Qonto-Idempotency-Key" in header for avoid duplicated request to API
- Add possibility to create POST request to the Qonto Api (see ApiClient.php)
- Update README with new requests for the ExternalTransfer.php

Commit:
Merge remote-tracking branch 'origin/master'
